bin/yb-ctl

```diff
@@ -694,10 +756,48 @@ class ClusterOptions:
         raise RuntimeError("No binary found for {}. Considered binary directories: {}".format(
             binary_name, binary_dirs))
 
-    def get_address(self, daemon_id, port_type, include_base_url=False):
-        port_str = str(self.base_ports[daemon_id.daemon_type][port_type])
+    def get_host_port(self, daemon_id, port_type):
         base_local_url = daemon_id.get_ip_address()
-        return port_str if not include_base_url else "{}:{}".format(base_local_url, port_str)
+        if port_type.endswith('_protocol'):
+            port_str = str(self.get_client_protocol_port(port_type.split('_')[0]))
+        else:
+            port_str = str(self.base_ports[daemon_id.daemon_type][port_type])
+        return "{}:{}".format(base_local_url, port_str)
+
+    def get_client_protocol_port(self, protocol_type):
```

Comment:
why not just update `self.base_ports` with `self.cluster_config` when we load it?